### ISSUE TITLE
[Webservice] Get order returns

### ISSUE THREAD

**Issue by TomanDesign** [PrestaShop/PrestaShop#17753]
I can't even find way to get order return list through prestashop API. Resource of type "order_returns" does not exists.

**Comment by khouloudbelguith** [comment]
Hi @TomanDesign,

Thanks for the report.
I have the same behavior as PS1.7.6.3.
Ping @PrestaShop/prestashop-product-team what do you think of this improvement?
Can we add `order_returns` to PrestaShop Webservice?

Thanks!

**Comment by LouiseBonnard** [comment]
@MatShir, is it planned in the rework of the order page? or for a second chapter? ;-)

**Comment by MatShir** [comment]
It is concerning the API, so it is not part of the order rework. We have a few issues with the WS categories. We could create an Epic.

**Comment by TomanDesign** [comment]
At this moment I have added 315 line:
`'order_returns' => array('description' => 'The Order returns', 'class' => 'OrderReturn'),`
after:
`'order_payments' => array('description' => 'The Order payments', 'class' => 'OrderPayment'),`
in \classes\webservice\WebserviceRequest.php

Now I can get list of returns, but I can't get product list.

**Comment by maicolcantagallo** [comment]
Hi, what is the state of this development? thanks

**Comment by matks** [comment]
I confirm this is something that should be done for the webservice so I pass it in `To Do`.

However as you can see there has been no development effort on the webservice for 1 years.

This does not mean we dont care, it's rather that we have so much topics to handle, this one cannot reach the top of our pile of things to do for now. And this is likely to stay this way for the coming months.

If you are interested in the topic feel free to invest yourself in it 😉  .

**Comment by Max84** [comment]
Hi @marionf , @matks
I need this feature. Can I do a PR?
I'm asking before starting to work because webservice will be refactored, if am I right.

**Comment by matks** [comment]
Hello @Max84 yes please go ahead ;)

We're working on a new API but I don't think it'll replace webservice, because it is an API that aims at allowing people to do back-office actions automatically. Webservice is more data-oriented.

Just one warning: if your PR modifies 999 lines of code in the webservice 😅 it might be a nightmare for us to review and test because the webservice is built in a... non-standard way. But I don't think it'll happen.

I think allowing the WS to handle order returns should be a fairly simple PR 😊 it is supposed to simply plugging the right things in place
